src/main/java/edu/kit/provideq/toolbox/api/ProblemDto.java

```diff
@@ -16,6 +18,7 @@ public class ProblemDto<InputT, ResultT> {
   private String typeId;
   private InputT input;
   private Solution<ResultT> solution;
+  private Optional<BoundWithInfo> bound;
```

Comment:
Can you remove the Optional type and add an `orElse(null)` to the assignment? It's going to be parsed into JSON anyway and similar to `Solution<ResultT>` it can also use null as fallback otherwise.
We want to have Optionals in the code logic, but we don't want to have Optionals in front facing data structures.